sql/catalyst/src/test/scala/org/apache/spark/sql/types/DecimalSuite.scala

```diff
@@ -228,4 +228,10 @@ class DecimalSuite extends SparkFunSuite with PrivateMethodTester {
     val decimal = Decimal.apply(bigInt)
     assert(decimal.toJavaBigDecimal.unscaledValue.toString === "9223372036854775808")
   }
+
+  test("SPARK-26038: toScalaBigInt/toJavaBigInteger not fitting long") {
```

Comment:
Should we test the branch when `decimalVal.ne(null)` is `true`?

Reply:
I just re-run `DecimalSuite` in the debugger, and haven't hit breakpoints in the branches. It seems there are no tests for nulls at all.